File: server/src/main/java/com/smartexpense/api/domain/model/ChatMessage.java
package com.smartexpense.api.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ChatMessage {
    private UUID id;
    private UUID userId;
    private String role; // 'USER' or 'AI'
    private String content;
    private LocalDateTime createdAt;
}

File: server/bin/main/com/smartexpense/api/domain/model/ChatMessage.kt
package com.smartexpense.api.domain.model

import java.time.LocalDateTime
import java.util.UUID

data class ChatMessage(
    val id: UUID? = null,
    val userId: UUID,
    val role: String,
    val content: String,
    val createdAt: LocalDateTime? = null
)
